### PR TITLE
add build.ps1 and update tests&docs

### LINKED ISSUE #3018: add PowerShell build script for C# template (for Windows)
# Feature Proposal

## Description

The AWS C# template (`aws-csharp`) includes a bash script for building and zipping the deploy package, `build.sh`. When developing on Windows, this is an inconvenience to use. I'd like to add a `build.ps1` PowerShell script to accomplish the same task. 

AFAIK, this is relatively common to offer `.cmd` or `.ps1`s alongside *nix scripts and executables, for example, in npm packages.

Similar or dependent issues:
* Supports #2858

### PR DESCRIPTION
**Pull request by FLGMwt**
## What did you implement:

Closes #3018 

## How did you implement it:

I added a `build.ps1` that mirrors the behavior of the `build.sh` bash script. Of note, the `ZipDirectory` method doesn't allow creating a zip in the file that it's zipping, so I create it in root and then copy it back to the publish.

## How can we verify it:

On a Windows machine, run

```
serverless create -t aws-csharp
./build.ps1
# note that a `deploy-package.zip` is created in `bin/Release/netcoreapp1.0/publish`
serverless deploy
# succeeds, deploys the package zip
serverless invoke -f hello
# returns "Hello World" response
```

## Todos:

- [x] Write tests
- [x] Write documentation
- [x] Fix linting errors
- [x] Make sure code coverage hasn't dropped
- [x] Provide verification config/commands/resources
- [x] Enable ["Allow edits from maintainers"](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) for this PR
- [x] Change ready for review message below

***Is this ready for review?:*** YES
***Is it a breaking change?:*** NO

### CODE REVIEW

File: docs/providers/aws/examples/hello-world/csharp/README.md [hunk 1]
```diff
@@ -27,6 +27,18 @@ zip -r ./deploy-package.zip ./*
 popd
```
Comment (FLGMwt) [inline docs/providers/aws/examples/hello-world/csharp/README.md]: Didn't want to rip it without asking, but with the build scripts in place, maybe this block adds noise?

Comment (mthenw) [inline docs/providers/aws/examples/hello-world/csharp/README.md]: Yeah, I think so. You can remove it and leave only reference to build scripts.

File: lib/plugins/create/templates/aws-csharp/build.ps1 [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+   $compressionLevel = [System.IO.Compression.CompressionLevel]::Optimal
+   [System.IO.Compression.ZipFile]::CreateFromDirectory($sourcedir,
+        $zipfilename, $compressionLevel, $false)
+}
+
+dotnet restore
+dotnet publish -c release
+if ($LASTEXITCODE -ne 0) { return }
+
+$publishDirectory = "bin/Release/netcoreapp1.0/publish"
```
Comment (mthenw) [inline lib/plugins/create/templates/aws-csharp/build.ps1]: Recently there was PR renaming "Release" to "release". I'm not C# expert. Which version is proper?

https://github.com/serverless/serverless/pull/3026

Comment (FLGMwt) [inline lib/plugins/create/templates/aws-csharp/build.ps1]: You're right, done!

**Pull request merged**